src/libcrun/ebpf.c

```diff
@@ -329,14 +332,36 @@ libcrun_ebpf_load (struct bpf_program *program, int dirfd, const char *pin, libc
 
       fd = bpf (BPF_PROG_LOAD, &attr, sizeof (attr));
       if (fd < 0)
-        return crun_make_error (err, errno, "bpf create %s", log);
+        return crun_make_error (err, errno, "bpf create `%s`", log);
+    }
+
+  if (pin)
+    {
+      union bpf_attr attr_open;
+
+      memset (&attr_open, 0, sizeof (attr_open));
+      attr_open.pathname = (uint64_t) pin;
+
+      replacefd = bpf (BPF_OBJ_GET, &attr_open, sizeof (attr_open));
+#ifdef BPF_F_REPLACE
+      if (replacefd < 0 && errno != ENOENT)
+        return crun_make_error (err, errno, "open `%s`", pin);
+#else
+      if (replacefd >= 0)
+        return crun_make_error (err, 0, "eBPF program already configured");
+ #endif
     }
 
   memset (&attr, 0, sizeof (attr));
   attr.attach_type = BPF_CGROUP_DEVICE;
   attr.target_fd = dirfd;
   attr.attach_bpf_fd = fd;
+#ifdef BPF_F_REPLACE
```

Comment:
Please add kernel requirement (v5.6) in the code comment
https://github.com/torvalds/linux/commit/7dd68b3279f1792103d12e69933db3128c6d416e

Reply:
thanks, I am going to rework the patch and handle better concurrent updates

Reply:
also, I'll have to move the check at runtime, not build time